backend/src/libs/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,57 @@
+import fp from 'fastify-plugin';
+
+import { ExceptionMessage } from '#libs/enums/enums.js';
+import { AuthError } from '#libs/exceptions/exceptions.js';
+import { ControllerHook } from '#libs/packages/controller/controller.js';
+import { HTTPCode, type HTTPMethod } from '#libs/packages/http/http.js';
+import { checkIsWhiteRoute } from '#libs/packages/server-application/server-application.js';
+import { type AuthTokenPayload, type JWTService } from '#packages/auth/auth.js';
+import { type UserService } from '#packages/users/users.js';
+
+type Options = {
+  services: {
+    userService: UserService;
+    jwtService: JWTService;
+  };
+};
+
+const authorization = fp<Options>((fastify, { services }, done) => {
+  fastify.decorateRequest('user', null);
+
+  fastify.addHook(ControllerHook.ON_REQUEST, async (request) => {
+    if (
+      checkIsWhiteRoute({
+        path: request.routerPath,
+        method: request.method as HTTPMethod,
+      })
+    ) {
+      return;
+    }
+
+    const [, token] = request.headers.authorization?.split(' ') ?? [];
+
+    if (!token) {
+      throw new AuthError({
+        message: ExceptionMessage.UNAUTHORIZED_USER,
+        status: HTTPCode.UNAUTHORIZED,
```

Comment:
Provide only message to custom errors

Reply:
@Liza-Veis suggested me removing the default value for status https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/39#discussion_r1301030732.

Do you recommend making a status non-changeable at all (only unauhorized)?

Reply:
https://bsa2023.slack.com/archives/C05P563E2MN/p1692617934865979?thread_ts=1692615709.704839&cid=C05P563E2MN

Reply:
should it look like so?
```ts
// auth plugin

throw new AuthError({message: ExceptionMessage.UNAUTHORIZED_USER})

// setErrorHandler in server application

if (error instanceof AuthError) {
  this.logger.error(`[Auth Error]: ${HTTPCode.UNAUTHORIZED} – ${error.message}`);

  const response: ServerCommonErrorResponse = {
    errorType: ServerErrorType.AUTHORIZATION,
    message: error.message,
  };

  return reply.status(HTTPCode.UNAUTHORIZED).send(response);
}
```

Reply:
Yep, but initErrorHandler method increases in large one, so please create a mapper(helper) that will return response and after execution of this one call logger and reply.status().send()

Reply:
let me clarify. I need to create a helper function that will have all `if` statements and return an appropriate response, right?

Reply:
@v0ldemar01, I tried to do what you asked. Please, could you tell me is it right?